packages/aws-cdk-lib/aws-codebuild/lib/fleet.ts

```diff
@@ -138,13 +196,37 @@ export class Fleet extends Resource implements IFleet {
       throw new Error('baseCapacity must be greater than or equal to 1');
     }
 
+    if (props.computeConfiguration) {
```

Comment:
I also had this validation to make sure the user saw how `ATTRIBUTE_BASED_COMPUTE` worked, leaving it up to you if you want to add it as well:

```suggestion
    if (props.computeType === FleetComputeType.ATTRIBUTE_BASED &&
      (!props.computeConfiguration || Object.keys(props.computeConfiguration).length === 0)) {
      throw new Error('At least one compute configuration criteria must be specified if computeType is "ATTRIBUTE_BASED"');
    }
    
    if (props.computeConfiguration) {
```